lib/CryptoLD.js

```diff
@@ -67,10 +67,63 @@ class CryptoLD {
     return this.suites.get(type).from(serialized);
   }
 
+  /**
+   * Imports a key pair instance via the provided `documentLoader` function,
+   * optionally checking it for revocation and required context.
+   *
+   * @param {string} keyId - Key ID or URI.
+   * @param {Function} documentLoader - JSON-LD Document Loader.
+   * @param {boolean} [checkContext=true] - Whether to check that the fetched
+   *   key document contains the context required by the key's crypto suite.
+   * @param {boolean} [checkRevoked=true] - Whether to check the key object for
+   *   the presence of the `revoked` timestamp.
+   *
+   * @returns {Promise<LDKeyPair>} Resolves with the appropriate key pair
+   *   instance.
+   */
+  async fromKeyId({
+    keyId, documentLoader, checkContext = true, checkRevoked = true
+  } = {}) {
+    if(!keyId) {
+      throw new TypeError('The "keyId" parameter is required.');
+    }
+    if(!documentLoader) {
+      throw new TypeError('The "documentLoader" parameter is required.');
+    }
+    let keyDocument;
+    try {
+      ({doc: keyDocument} = await documentLoader(keyId));
+    } catch(e) {
+      const error = new Error('Error fetching document: ' + e.message);
+      error.cause = e;
+      throw error;
+    }
+    const fetchedType = keyDocument.type;
```

Comment:
I was expecting these checks to happen in derived classes so they can have full control. I guess this is fine as a default implementation which they can override as needed -- but it's awkward because the type is being checked *before* the context is checked. It's the `@context` that governs how the type will appear.

Seeing now how all this works in crypto-ld exposes some odd ergonomics -- but I don't have any good simple fixes to suggest.

So, this is going to be a little bit backwards, but maybe that's ok. A comment like this may help future devs a bit:

```suggestion
    /* Note: `type` is presumed to be an alias in all LD keys here
    as the `@context` that would define it is checked *after* this
    point. LD suite authors should follow convention to avoid any
    issues here. If framing is required, another class that extends
    CryptoLd will need to be used to provide that functionality. */
    const fetchedType = keyDocument.type;
```

Reply:
So, I'm curious to hear more about the odd ergonomics. Tell me more - what's your concern?
Like I mentioned in the [comment on the 2018 pr](https://github.com/digitalbazaar/ed25519-verification-key-2018/pull/14#discussion_r605297903), I'm not sure doing the checks in derived classes would be at all useful. (It would be a cut & paste of exactly the same code.) Is that not the case?

As for whey the `type` is checked before `@context` -- it's because the `type` is needed to select the suite from the installed suites, which only _then_ gives you the context to check.

Reply:
What I can do, in case it's useful down the line, is to rearrange it a bit so that it's override-able in individual suites.

Reply:
> Tell me more - what's your concern?

Well, fetching the property `type` before you even know its definition is what's weird. The `@context` defines that term (or doesn't!). But we don't determine the suite based on the `@context` (and maybe also the type), we determine it solely based on the type value ... which we get from an unknown property. So we're assuming the property is ok.

To give a contrived example that wouldn't work:

```js
{
  "@context": "my-super-new-context-that-defines-type-to-mean-favorite-old-suite",
  "type": "Ed25519VerificationKey2018",
  "@type": "MySuperAwesomeSuite2150",
  "..."
}
```

A less contrived example may feature a `@context` that doesn't define the `type` alias -- for some important future reason. I'm ok with what we have, but this is why it's a bit backwards/awkward. I think keying off the `@context` would be more resilient. We can think about this in the future.

Reply:
@dlongley I think I see what you mean. Except.. a whole lot of our crypto infrastructure depends on an un-aliased `type` property being present. So, possibly too late for that.

Reply:
All that said, I'm happy to amend `cryptoLD.use()` to also be able to key off of `@context`.

Reply:
@dmitrizagidulin, to be clear, our crypto infrastructure did *not* overly depend on `type` being present in untrusted data for many years. We're only now removing a lot of this extra `framing` stuff that dealt with it -- because you don't need it in trusted scenarios.

It could be that some of reliance on `type` has leaked in very recently (I'm not even sure that's true) because people did not understand that LD security is greater in scope than how suites are represented in JSON/DID Documents. But, also, as mentioned before, most of the APIs in our infrastructure are relying on trusted data with an expected format -- and this call is the first (?) API that accepts untrusted data -- and it may be the first time we're introducing an untrusted area that depends on `type` in this way.

So I don't actually think the existing infrastructure has the historical property you're describing here and that's why I'm raising a red (orangish?) flag here that we should be careful with this introduction.